src/cli/upload.rs

```diff
@@ -125,3 +159,41 @@ fn do_upload(buffer: Vec<u8>, asset_id: u64, cookie: &str) -> anyhow::Result<()>
 
     Ok(())
 }
+
+/// Implementation of do_upload that supports the new open cloud api.
+/// I'm sure there's a better of doing this. Please correct it if so.
+/// see https://developer.roblox.com/en-us/articles/open-cloud
+fn do_upload_open_cloud(buffer: Vec<u8>, universe_id: u64, asset_id: u64, api_key: &str) -> anyhow::Result<()> {
+    let url = format!(
+        "https://apis.roblox.com/universes/v1/{}/places/{}/versions?versionType=Published",
+		universe_id,
+        asset_id
+    );
+
+    let client = reqwest::Client::new();
+
+    let build_request = move || {
+        client
+            .post(&url)
+            .header("x-api-key", api_key)
+            .header(CONTENT_TYPE, "application/xml")
+            .header(ACCEPT, "application/json")
+            .body(buffer.clone())
```

Comment:
Why clone the buffer? Looks like you don't use it again.

Reply:
That was actually copied and modified from the do_upload function that already exists. In the case of do_upload, this is required for the CSRF challenge. I've pushed a commit that will remove this from do_upload_open_cloud.